src/lib/support/LifetimePersistedCounter.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "LifetimePersistedCounter.h"
+
+#include <platform/PersistedStorage.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+namespace chip {
+
+LifetimePersistedCounter::LifetimePersistedCounter() : mId(chip::Platform::PersistedStorage::kEmptyKey), mNextEpoch(0) {}
+
+LifetimePersistedCounter::~LifetimePersistedCounter() {}
+
+CHIP_ERROR
+LifetimePersistedCounter::Init(const chip::Platform::PersistedStorage::Key aId)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // Store the ID.
+    mId = aId;
+    uint32_t startValue;
+
+    // Read our previously-stored starting value.
+    err = ReadStartValue(startValue);
+    SuccessOrExit(err);
+
+    err = PersistNextEpochStart(startValue);
```

Comment:
this seems to write the same value that was read... could we remove this line?

Reply:
removed